app.js

```diff
@@ -78,72 +69,50 @@ app.get("/*", (req, res) => {
 
   lookup(decodeURI(slug)).then(
     (destination) => {
-      const resultURL = destination
       var resultQuery = combineQueries(
-        querystring.parse(new URL(destination).searchParams.toString()),
+        querystring.parse(new URL(destination).search),
         query
       );
-      const parsedDestination = new URL(destination)
-      const url = parsedDestination.origin + parsedDestination.pathname + resultQuery + parsedDestination.hash
-      console.log({ url })
-      res.redirect(302, url)
+      const parsedDestination = new URL(destination);
+      const finalURL = parsedDestination.origin + parsedDestination.pathname + resultQuery + parsedDestination.hash
+      console.log({ finalURL })
+      res.redirect(302, finalURL)
     },
     (error) => {
-      if (error == 404) {
-        res.redirect(302, "https://goo.gl/" + slug);
-      } else {
-        res.status(error);
-      }
+      res.redirect(302, "https://hackclub.com/404");
```

Comment:
Is Hack Club using analytics? You should set up a separate referrer/tracking campaign for any hack.af-induced 404s so you can look for broken links that is being placed around the internet. 

Something like

`https://hackclub.com/404?utm_source=hack.af&utm_campaign=[slug]`

Reply:
ATM there is no analytics that I know of, I will keep this on my to-do and make either just log it on a hack.af dashboard side itself on the next PR.